wai-extra/Network/Wai/Parse.hs

```diff
@@ -213,24 +288,56 @@ conduitRequestBody _ UrlEncoded rbody add = do
 conduitRequestBody backend (Multipart bound) rbody add =
     parsePieces backend (S8.pack "--" `S.append` bound) rbody add
 
+conduitRequestBodyEx :: ParseRequestBodyOptions
+                     -> BackEnd y
+                     -> RequestBodyType
+                     -> IO S.ByteString
+                     -> (Either Param (File y) -> IO ())
+                     -> IO ()
+conduitRequestBodyEx o _ UrlEncoded rbody add = do
+    -- NOTE: in general, url-encoded data will be in a single chunk.
+    -- Therefore, I'm optimizing for the usual case by sticking with
+    -- strict byte strings here.
+    size <- newIORef 0
+    let loop front = do
```

Comment:
Is there any reason to use the mutable reference approach here? How about just including `cursize` as a parameter to `loop`?